File: Session-3/DDD-Sample-codes/rtgs-return-management-springboot/src/main/java/com/bank/rtgs/returnmanagement/application/dto/RtgsReturnResponse.java
package com.bank.rtgs.returnmanagement.application.dto;

import com.bank.rtgs.returnmanagement.domain.model.RtgsReturn;
import com.bank.rtgs.returnmanagement.domain.model.ReturnStatus;
import java.time.Instant;
import java.util.UUID;

public record RtgsReturnResponse(
        UUID id,
        String originalTransactionReference,
        String reasonCode,
        String reasonDescription,
        ReturnStatus status,
        String payload,
        Instant createdAtUtc,
        Instant generatedAtUtc,
        Instant sentAtUtc,
        Instant acknowledgedAtUtc
) {
    public static RtgsReturnResponse from(RtgsReturn rtgsReturn) {
        return new RtgsReturnResponse(
                rtgsReturn.getId(),
                rtgsReturn.getOriginalTransactionReference().value(),
                rtgsReturn.getReason().code(),
                rtgsReturn.getReason().description(),
                rtgsReturn.getStatus(),
                rtgsReturn.getPayload(),
                rtgsReturn.getCreatedAtUtc(),
                rtgsReturn.getGeneratedAtUtc(),
                rtgsReturn.getSentAtUtc(),
                rtgsReturn.getAcknowledgedAtUtc()
        );
    }
}

File: Session-3/DDD-Sample-codes/rtgs-return-management-dotnet10/Application/Dtos/RtgsReturnResponse.cs
using Rtgs.ReturnManagement.Api.Domain.Model;

namespace Rtgs.ReturnManagement.Api.Application.Dtos;

public sealed record RtgsReturnResponse(
    Guid Id,
    string OriginalTransactionReference,
    string ReasonCode,
    string ReasonDescription,
    ReturnStatus Status,
    string? Payload,
    DateTime CreatedAtUtc,
    DateTime? GeneratedAtUtc,
    DateTime? SentAtUtc,
    DateTime? AcknowledgedAtUtc
)
{
    public static RtgsReturnResponse From(RtgsReturn rtgsReturn) => new(
        rtgsReturn.Id,
        rtgsReturn.OriginalTransactionReference.Value,
        rtgsReturn.Reason.Code,
        rtgsReturn.Reason.Description,
        rtgsReturn.Status,
        rtgsReturn.Payload,
        rtgsReturn.CreatedAtUtc,
        rtgsReturn.GeneratedAtUtc,
        rtgsReturn.SentAtUtc,
        rtgsReturn.AcknowledgedAtUtc
    );
}
